### DISCUSSION TITLE
Improved invoices and more timely usage data

### DISCUSSION THREAD

**Discussion by kevcodez** [supabase/supabase#28480] [Changelog]
Currently, usage data on the invoice breakdown and [organization usage page](https://supabase.com/dashboard/org/_/usage) has a 24-hour delay. Starting from **August 26th**, the usage data will have no more of 1 hour delay for new customers. Afterwards, the changes will be rolled out to existing customer gradually. We're also working on additional improvements to provide better usage insights.

<img width="809" alt="Screenshot 2024-07-31 at 21 06 58" src="https://github.com/user-attachments/assets/5365a59b-d49c-42de-a5cd-a2cd7f301b8a">

-----

Additionally, we are revamping invoices to provide more detailed breakdowns of usage for enhanced transparency. Due to our new proration of project add-ons and storage down to the hour, you may notice slight variances in your monthly bill. For the majority of line items, you’ll see the project reference and usage on the invoice, which should make it clearer which project allocated the usage/costs.

A few examples:

Compute Hours is broken down per project and the compute credits ($10) is displayed as discount for the compute line item.
<img width="1187" alt="Screenshot 2024-08-08 at 20 34 47" src="https://github.com/user-attachments/assets/135a896c-c427-4ce9-8e22-aa96003376b6">

Egress is broken down to each project and displays included quota (250GB) and over-age pricing ($0.09/GB)
<img width="1175" alt="Screenshot 2024-08-08 at 20 34 57" src="https://github.com/user-attachments/assets/bb0afcf5-4b7f-43bf-a33f-052052b86091">

Realtime Messages line item shows package-based pricing with $2.50 per million.
<img width="1183" alt="Screenshot 2024-08-08 at 20 35 26" src="https://github.com/user-attachments/assets/64699237-8638-4c4d-b40c-b06003ff5b1f">

**Comment by cloetensbrecht** [comment]
Is it possible to change the project reference? For internal followup it would be better that we can see the name of the project instead of the current reference key.

**Reply by kevcodez** [reply]
The name of the project is not stable and can change during a billing cycle, so we have opted to use the project id (reference) on the invoice for breaking usage down, you can use the reference to open any project via https://supabase.com/dashboard/project/<project-id>

Invoice line items are grouped on event ingestion time, so in case of a name change, it would group usage under the new name instead of having all project events under the same sub-item.

You can also filter usage by a project in your [organization's usage page](https://supabase.com/dashboard/org/_/usage).
